backend/mora/service/manager.py

```diff
@@ -13,20 +13,145 @@
 
 """
 import uuid
+import flask
 
 from . import address
 from . import handlers
+from . import orgunit
+from . import facet
+from . import employee
 from .. import common
 from .. import lora
 from .. import mapping
 from .. import util
 from .. import validator
 
+SEARCH_FIELDS = {
+    'e': 'tilknyttedebrugere',
+    'ou': 'tilknyttedeenheder'
+}
 
-class ManagerRequestHandler(handlers.OrgFunkRequestHandler):
+
+class ManagerRequestHandler(
+    handlers.OrgFunkRequestHandler,
+    handlers.ReadingRequestHandler
+):
+
+    __slots__ = ()
     role_type = 'manager'
     function_key = mapping.MANAGER_KEY
 
+    @classmethod
+    def has(cls, scope, objid):
+        pass
+
+    @classmethod
+    def get(cls, c, type, objid):
+
+        search = {
+            SEARCH_FIELDS[type]: objid
+        }
+
+        function_effects = [
+            cls.get_one_mo_object(effect, start, end, funcid)
+            for funcid, funcobj in c.organisationfunktion.get_all(
+                funktionsnavn=cls.function_key,
+                **search,
+            )
+            for start, end, effect in c.organisationfunktion.get_effects(
+                funcobj,
+                {
+                    'relationer': (
+                        'opgaver',
+                        'adresser',
+                        'organisatoriskfunktionstype',
+                        'tilknyttedeenheder',
+                        'tilknyttedebrugere',
+                        'tilknyttedefunktioner',
+                    ),
+                    'tilstande': (
+                        'organisationfunktiongyldighed',
+                    ),
+                },
+                {
+                    'attributter': (
+                        'organisationfunktionegenskaber',
+                    ),
+                    'relationer': (
+                        'tilhoerer',
+                        'tilknyttedeorganisationer',
+                        'tilknyttedeitsystemer',
+                    ),
+                },
+            )
+            if util.is_reg_valid(effect)
+        ]
+        return flask.jsonify(function_effects)
+
+    @classmethod
+    def get_one_mo_object(cls, effect, start, end, funcid):
+        c = common.get_connector()
+
+        persons = list(mapping.USER_FIELD.get_uuids(effect))
+        manager_types = list(mapping.ORG_FUNK_TYPE_FIELD.get_uuids(effect))
+        manager_levels = list(mapping.MANAGER_LEVEL_FIELD.get_uuids(effect))
+        addresses = list(mapping.FUNCTION_ADDRESS_FIELD.get_uuids(effect))
+        responsibilities = list(mapping.RESPONSIBILITY_FIELD.get_uuids(effect))
+        org_units = list(mapping.ASSOCIATED_ORG_UNIT_FIELD.get_uuids(effect))
+
+        func = {
+            mapping.UUID: funcid,
+            mapping.RESPONSIBILITY: [
```

Comment:
You should probably sort these by name using `sorted()`; that may be why you needed to change the test.

Reply:
Jeg sorterer - Og de får lov at stå sorteret i testen.

Reply:
Done!